assets/src/modules/BaseLayersMap.js

```diff
@@ -0,0 +1,399 @@
+    }
+
+    get baseLayersGroup(){
+        return this._baseLayersGroup;
+    }
+
+    get overlayLayersAndGroups(){
+        return this._overlayLayersAndGroups;
+    }
+
+    // Get overlay layers (not layerGroups)
+    get overlayLayers(){
+        return this._overlayLayersGroup.getLayersArray();
+    }
+
+    get overlayLayersGroup(){
+        return this._overlayLayersGroup;
+    }
+
+    /**
+     * Synchronize new OL view with OL2 one
+     * @memberof Map
+     */
+    syncNewOLwithOL2View(){
+        this.getView().animate({
+            center: mainLizmap.center,
+            zoom: mainLizmap.lizmap3.map.getZoom(),
+            duration: 0
+        });
+    }
+
+    changeBaseLayer(name){
+        let selectedBaseLayer;
+        // Choosen base layer is visible, others not
+        this.baseLayersGroup.getLayers().forEach( baseLayer => {
+            if (baseLayer.get('name') == name) {
+                selectedBaseLayer = baseLayer;
+                baseLayer.set("visible", true, true);
+            } else {
+                baseLayer.set("visible", false, true);
+            }
+        });
+
+        this._baseLayersGroup.changed();
+
+        // If base layer projection is different from project projection
+        // We must set the project extent to the View to reproject nicely
+        if (selectedBaseLayer?.getSource().getProjection().getCode() !== mainLizmap.projection) {
+            this.getView().getProjection().setExtent(mainLizmap.lizmap3.map.restrictedExtent.toArray());
+        } else {
+            this.getView().getProjection().setExtent(getProjection(mainLizmap.projection).getExtent());
+        }
+
+        // Trigger event
+        lizMap.events.triggerEvent("lizmapbaselayerchanged", { 'layer': name});
+
+        // Refresh metadatas if sub-dock is visible
+        if ( document.getElementById('sub-dock').offsetParent !== null ) {
+            lizMap.events.triggerEvent("lizmapswitcheritemselected", {
+                'name': name, 'type': 'baselayer', 'selected': true
+            });
+        }
+    }
+
+    getActiveBaseLayer(){
+        return this._baseLayersGroup.getLayers().getArray().find(
+            layer => layer.getVisible()
+        );
+    }
+
+    /**
+     * Return overlay layer if `name` matches.
+     * `name` is unique for every layers
+     */
+    getLayerByName(name){
+        return this.overlayLayers.find(
+            layer => layer.get('name') === name
+        );
+    }
+
+    /**
+     * Return overlay layer or group if `name` matches.
+     * `name` is unique for every layers/groups
+     */
+    getLayerOrGroupByName(name){
+        return this.overlayLayersAndGroups.find(
+            layer => layer.get('name') === name
+        );
+    }
+
+    /**
+     * Return overlay layer if `typeName` matches
+     */
+    getLayerByTypeName(typeName){
+        return this.overlayLayers.find(
+            layer => layer.getSource().getParams?.()?.LAYERS === typeName
+        );
+    }
```

Comment:
The `LAYERS` is not the typeName, it is a wmsName.

Reply:
I'll fix it